browser-sign-in/OktaBrowserSignIn/AppDelegate.swift

```diff
@@ -15,28 +15,78 @@
  */
 
 import UIKit
-import OktaOidc
+import WebAuthenticationUI
 
 @UIApplicationMain
 class AppDelegate: UIResponder, UIApplicationDelegate {
     
     var window: UIWindow?
-
+    var configForUITests: [String: String]? {
+        let env = ProcessInfo.processInfo.environment
+        guard let oktaURL = env["OKTA_URL"], oktaURL.count > 0,
+              let clientID = env["CLIENT_ID"],
+              let redirectURI = env["REDIRECT_URI"],
+              let logoutRedirectURI = env["LOGOUT_REDIRECT_URI"] else {
+            return nil
+        }
+        return ["issuer": "\(oktaURL)/oauth2/default",
+                "clientId": clientID,
+                "redirectUri": redirectURI,
+                "logoutRedirectUri": logoutRedirectURI,
+                "scopes": "openid profile offline_access"
+        ]
+    }
+    
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        self.window = UIWindow(frame: UIScreen.main.bounds)
+        
+        if ProcessInfo.processInfo.arguments.contains("--reset-keychain") {
+            try? Keychain.Search().delete()
+        }
+        window?.rootViewController = self.getRootViewController()
+        window?.makeKeyAndVisible()
```

Comment:
`self.window` is Optional, but we know we have an instance.  So in cases like this, I find it nice to assign the window to a local variable so we don't have to unwrap the optional.

```suggestion
        if ProcessInfo.processInfo.arguments.contains("--reset-keychain") {
            try? Keychain.Search().delete()
        }

       let window = UIWindow(frame: UIScreen.main.bounds)
        window.rootViewController = self.getRootViewController()
        window.makeKeyAndVisible()
        self.window = window
```